### PR TITLE
Add Apple M1 Config on spir-v

### PR DESCRIPTION
**Pull request by raikonenfnu**
-Add tuned apple M1 Config for spirv backend

### CODE REVIEW

File: iree/compiler/Codegen/SPIRV/AppleConfig.cpp [hunk 1]
```diff
@@ -0,0 +1,50 @@
+// Copyright 2021 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+//===- AppleConfig.h - Apple CodeGen Configurations ---------------------===//
```
Comment (antiagainst) [inline iree/compiler/Codegen/SPIRV/AppleConfig.cpp:7]: Nit: expand this to take 80 chars.

File: iree/compiler/Codegen/SPIRV/AppleConfig.cpp [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+//
+// This file contains CodeGen configurations for Apple M1 GPUs.
```
Comment (antiagainst) [inline iree/compiler/Codegen/SPIRV/AppleConfig.cpp:9]: Just say "Apple GPUs". We'll have more Apple GPU supported later.

Comment (antiagainst) [inline iree/compiler/Codegen/SPIRV/AppleConfig.cpp:9]: Just say "Apple GPUs". We'll have more Apple GPU architectures supported later.

File: iree/compiler/Codegen/SPIRV/KernelConfig.cpp [hunk 3]
```diff
@@ -543,6 +543,9 @@ static LogicalResult setSPIRVOpConfig(const spirv::TargetEnv &targetEnv,
     case spirv::Vendor::Qualcomm:
       result = detail::setAdrenoCodeGenConfig(targetEnv, rootOp);
       break;
+    case spirv::Vendor::Apple:
```
Comment (antiagainst) [inline iree/compiler/Codegen/SPIRV/KernelConfig.cpp:546]: Please sort this alphabetically.

Comment (antiagainst) [inline iree/compiler/Codegen/SPIRV/KernelConfig.cpp:546]: Please sort this alphabetically.

**Review by antiagainst — APPROVED** [review]
LGTM! Just a few nits.

**Comment by raikonenfnu** [comment]
Done! thanks for the comments :)

File: iree/compiler/Codegen/SPIRV/KernelConfig.cpp [hunk 4]
```diff
@@ -534,6 +534,8 @@ static LogicalResult setSPIRVOpConfig(const spirv::TargetEnv &targetEnv,
     case spirv::Vendor::AMD:
       result = detail::setAMDCodeGenConfig(targetEnv, rootOp);
       break;
+    case spirv::Vendor::Apple:
+      result = detail::setAppleCodeGenConfig(targetEnv, rootOp);
```
Comment (antiagainst) [inline iree/compiler/Codegen/SPIRV/KernelConfig.cpp:538]: `break` is missing here. This will fall to the Mail configuration later.

**Review by antiagainst — CHANGES_REQUESTED** [review]

File: iree/compiler/Codegen/SPIRV/KernelConfig.cpp (hunk 4 above)
Comment (raikonenfnu) [reply iree/compiler/Codegen/SPIRV/KernelConfig.cpp:538]: oops, sorry yeah, good catch :)

**Review by antiagainst — APPROVED** [review]

**Comment by antiagainst** [comment]
Nice, thanks @raikonenfnu for adding this!

**Pull request merged**